sdk/aqueduct/decorator.py

```diff
@@ -208,21 +208,42 @@ def _type_check_decorated_function_arguments(
                 )
 
 
-def _convert_argument_to_parameter(
-    *input_artifacts: Any, function_argument_names: List[str]
+def _convert_input_arguments_to_parameters(
+    *input_artifacts: Any, func_params: Mapping[str, inspect.Parameter]
 ) -> List[BaseArtifact]:
     """
     Converts non-artifact inputs to parameters.
+
+    Errors if the implicitly-created parameter collides with an existing one. Also errors if the function has a
+    variable-length parameter, since we don't know what name to attribute to those.
+
+    `func_params` maps from parameter name to an `inspect.Parameter` object containing additional information.
     """
+    # KEYWORD_ONLY parameters are allowed, since they are guaranteed to have a name.
+    # Note that we only accept them after "*" arguments, since we error out on VAR_POSITIONAL (eg. *args).
```

Comment:
Not sure I understand what this sentence means. Are you saying if we have `foo(*args, param)`, then we support implicitly creating parameter for `param`? Then it seems to contradict with line 225, since we have `*`, `implicit_params_disallowed` will be true, which prevents the creation of implicit parameters.

Reply:
ah, so `*args` is different from `*` (I know, right). If you have `foo(*, param)`, then `param` will be of type `KEYWORD_ONLY`, which is ok! `implicit_params_disallowed` will not be true in such a case, and we can create the param implicitly. The `*` wildcard is just there to force any arguments behind it to be keyword arguments (https://peps.python.org/pep-3102/).


I can add https://peps.python.org/pep-0362/#parameter-object to the comment

Reply:
Since we have this line:
```
  implicit_params_disallowed = any(
      param.kind in disallowed_kinds for param in func_params.values()
  )
```
So for `foo(*, param)`, won't `*` make `implicit_params_disallowed` to be true due to the use of `any`?

Reply:
I guess I'm not super familiar with what `*` means in `foo(*, param)`. If I call this function with (x, y), where x is a non-artifact and y is an artifact, we will attempt to create an implicit parameter for x, and since x doesn't have a name (because the first argument to `foo` is `*`), things will fail? Maybe we should grab a time to chat online to clarify this. I'm also having some trouble understanding the diff between `foo(*, param)` and `foo(*args, param)`. :)